flask/helpers.py

```diff
@@ -575,12 +585,21 @@ def send_file(filename_or_fp, mimetype=None, as_attachment=False,
             warn('Access %s failed, maybe it does not exist, so ignore etags in '
                  'headers' % filename, stacklevel=2)
 
-        if conditional:
+    if conditional:
+        try:
+            # Added in werkzeug 0.12
+            rv = rv.make_conditional(request, accept_ranges=True,
+                                     complete_length=fsize)
+        except TypeError:
```

Comment:
This seems a bit broad. Could we change this to a version check for Werkzeug, or check for existence of a function like done in the tests?